### ISSUE TITLE
Ability to configure compression type for no-dictionary columns

### ISSUE THREAD

**Issue by mayankshriv** [apache/incubator-pinot#5407]
No-dictionary dimension columns are Snappy compressed, and metrics are not compressed. While the SegmentGenerationConfig has `_rawIndexCompressionType` to configure the compression type, this is not available in the indexing config. We need to enhance indexing config such that we are able to configure various parameters for no-dictionary columns:
1. Compression to use for the column.
2. Num docs per chunk (and other such params for advanced users).

**Comment by kishoreg** [comment]
We should move towards the FieldConfig mechanism that was added by @siddharthteotia. Its nice to have all configs related to a field in one place.

**Comment by mayankshriv** [comment]
Yes, agree. The existing mechanism is just getting bloated with the way new fields are added.

**Comment by xiangfu0** [comment]
Closing — this has been implemented. Multiple compression codecs are now supported for no-dictionary columns: LZ4 (PR #6804), Zstandard (PR #6876), plus GZIP, SNAPPY, and CLP codecs added in subsequent PRs.
